### ISSUE TITLE
New API & filtering by token on getMarket

### ISSUE THREAD

**Issue by nicholasball** [etherdelta/etherdelta.github.io#209]
Hi,

I seem to be having an issue with the getMarket call when I provide a token address. Note that in the docs that user address and token address are optional so I leave out the user address.

However, there doesn't seem to be any filtering being done as I receive orders and trades for all tokens. Is this a known issue or a problem with my implementation?

Many thanks!

**Comment by zackcoburn** [comment]
Yes, the "orders" and "trades" events that get emitted continuously in real
time will have orders and trades for all tokens. You'd then have to filter
them on your end. Only the initial "getMarket" result will be filtered.

Zack Coburn

NOTICE: This e-mail transmission (and/or the attachments accompanying it)
may contain confidential information belonging to the sender which is
protected by law. The information is intended only for the use of the
intended recipient. If you are not the intended recipient, you are hereby
notified that any disclosure, copying, distribution or the taking of any
action in reliance on the contents of this information is strictly
prohibited. Any unauthorized interception of this transmission is illegal.
If you have received this transmission in error, please promptly notify the
sender by reply e-mail, and then destroy all copies of the transmission.

**Comment by nicholasball** [comment]
OK, thanks for the confirmation.

Do you plan on filtering the getMarket based on the given input? Not sure what the point is of sending the token address if you are getting a stream of everything.

**Comment by zackcoburn** [comment]
Well, the "market" response to "getMarket" should be filtered (with the
exception of the "returnTicker" part, which shows all tokens). The
"trades," "orders," "funds," "myTrades," "myOrders," "myFunds" events that
get emitted are all unfiltered (they start coming even if you don't call
"getMarket"). I can consider providing an option that filters these events
if you'd like.

Zack Coburn

NOTICE: This e-mail transmission (and/or the attachments accompanying it)
may contain confidential information belonging to the sender which is
protected by law. The information is intended only for the use of the
intended recipient. If you are not the intended recipient, you are hereby
notified that any disclosure, copying, distribution or the taking of any
action in reliance on the contents of this information is strictly
prohibited. Any unauthorized interception of this transmission is illegal.
If you have received this transmission in error, please promptly notify the
sender by reply e-mail, and then destroy all copies of the transmission.

**Comment by bitcoinmeetups** [comment]
@Zack, that notice which is attached to every post you make is pretty long and distracting. Any way to disable it?

**Comment by zackcoburn** [comment]
Maybe we'll let you dismiss it. It's on the list.

Zack Coburn

NOTICE: This e-mail transmission (and/or the attachments accompanying it)
may contain confidential information belonging to the sender which is
protected by law. The information is intended only for the use of the
intended recipient. If you are not the intended recipient, you are hereby
notified that any disclosure, copying, distribution or the taking of any
action in reliance on the contents of this information is strictly
prohibited. Any unauthorized interception of this transmission is illegal.
If you have received this transmission in error, please promptly notify the
sender by reply e-mail, and then destroy all copies of the transmission.
